src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeCleanup;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.CodeAnalysis.Rename.ConflictEngine;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Rename
+{
+    /// <summary>
+    /// Equivalent to <see cref="HeavyweightRenameLocations"/> except that references to symbols are kept in a lightweight fashion
+    /// to avoid expensive rehydration steps as a host and OOP communicate.
+    /// </summary>
+    internal sealed partial class LightweightRenameLocations
+    {
+        public readonly ISymbol Symbol;
```

Comment:
Are we worried about hydration of this symbol?

Reply:
it's technically not hydrated or involved in marshalling at all.  but i can see how this is confusing.  I can do an immediate followup to remove this.  It will require more passing around of data as parameters.  but htat's likely a good thing.

Reply:
done in followup: https://github.com/dotnet/roslyn/pull/62803